tests/utils.py

```diff
@@ -33,55 +33,58 @@ def run_privileged_container(
     image: Image,
     command: List[str],
     volumes: Dict[str, Dict[str, str]] = None,
-    auto_remove: bool = True,
     **extra_kwargs: Any,
-) -> Tuple[Optional[Container], str]:
+) -> str:
     if volumes is None:
         volumes = {}
-    container_or_logs = docker_client.containers.run(
-        image,
-        command,
-        privileged=True,
-        network_mode="host",
-        pid_mode="host",
-        userns_mode="host",
-        volumes=volumes,
-        auto_remove=auto_remove,
-        stderr=True,
-        **extra_kwargs,
-    )
-    if isinstance(container_or_logs, Container):
-        container, logs = container_or_logs, container_or_logs.logs().decode()
-    else:
-        assert isinstance(container_or_logs, bytes), container_or_logs
-        container, logs = None, container_or_logs.decode()
+
+    container = None
+    try:
+        container = docker_client.containers.run(
+            image,
+            command,
+            privileged=True,
+            network_mode="host",
+            pid_mode="host",
+            userns_mode="host",
+            volumes=volumes,
+            stderr=True,
+            detach=True,
+            **extra_kwargs,
+        )
+        # let it finish
+        exit_status = container.wait()["StatusCode"]
+        # and read its log
+        logs = container.logs(stdout=True, stderr=True)
+
+        if exit_status != 0:
+            raise ContainerError(container, exit_status, command, image, logs)
+    finally:
+        if container is not None:
+            container.remove()
```

Comment:
you can pass `rm=Tru` to `docker_client.containers.run

Reply:
but then I won't have logs